src/math_parser_diag.cpp

```diff
@@ -1321,6 +1321,55 @@ std::function<double( dialogue & )> vision_range_eval( char scope,
     };
 }
 
+std::function<double( dialogue & )> calories_eval( char scope,
+        std::vector<diag_value> const &/* params */, diag_kwargs const &kwargs )
+{
+    diag_value format_value( std::string( "raw" ) );
+    if( kwargs.count( "format" ) != 0 ) {
+        format_value = *kwargs.at( "format" );
+    }
+
+    return[format_value, beta = is_beta( scope )]( dialogue const & d ) {
+        std::string format = format_value.str( d );
+        if( format != "raw" && format != "percent" ) {
+            debugmsg( R"(Unknown format type "%s" for calories, assumning "raw")", format );
+            format = "raw";
+        }
+
+        if( format == "percent" ) {
+            if( d.actor( beta )->get_character() ) {
+                int divisor = d.actor( beta )->get_healthy_kcal() / 100;
```

Comment:
Do floating point math with this, all EOC math is.
```suggestion
                double divisor = d.actor( beta )->get_healthy_kcal() / 100.0;
```